lms/djangoapps/courseware/access.py

```diff
@@ -493,7 +493,6 @@ def can_load():
 
         return (
             _visible_to_nonstaff_users(descriptor) and
-            _can_access_descriptor_with_milestones(user, descriptor, course_key) and
```

Comment:
Access implications:
* What is the latest on the mobile access story?  Will that be fixed independently of these changes?
* This check here also prevented someone from accessing the gated block directly (not via the LMS courseware). This includes LTI (which iframes the xblock) and mobile (which displays the xBlock via a webview).  They both access the xBlock via [the render_xblock View](https://github.com/edx/edx-platform/blob/5750495f86cd83985b4793416e981cf91d52cdf6/lms/djangoapps/courseware/views/views.py#L1407).  To check, go to the following url on your sandbox: courses.edx.org/xblock/<USAGE_KEY_OF_GATED_BLOCK>

Reply:
Currently there is a bug in the way mobile handles gated content.  @staubina made the following ticket to track it: https://openedx.atlassian.net/browse/LEARNER-3794

Since our changes don't change the current behavior and that issue needs to be fixed independently anyway I don't think the mobile bug should block our deployment.

Reply:
I think the mobile bug is that the gated content appears in the course outline.  However, this particular line in the code should have prevented the learner from actually accessing that gated content.

Reply:
@nasthagiri 
I think the use case you mention above, user accessing gated block directly should be covered, as the render of the xblock won't show the content if a preq exists and hasn't been made. I tested on the sandbox by going to:
https://whitelabel.sandbox.edx.org/xblock/block-v1:bfillerX+BF101+2017_12+type@sequential+block@3e73b2eeab81434faf378cba9c519f6a

And got the "Content Locked" page.
![xblock-direct](https://user-images.githubusercontent.com/27503539/34728240-60024886-f527-11e7-9c68-8c2747951c5d.png)

Reply:
I can quickly verify this.  Can you send me a link to a course on prod that has gated content, if you have it readily available?

Reply:
@bill-filler What if the learner goes directly to an xBlock within the sequential though?   (That's what the LTI/Mobile view allows.)  Do they still get that message?  If they do, then that works!

Reply:
@nasthagiri 
Not sure how I would test if learner goes directly to an xBlock within the sequential?
Here is a course on prod that has gated content:
https://courses.edx.org/courses/course-v1:MITx+CTL.SC3x+1T2018/course/

Reply:
here's one in stage to play around with https://globalacademy-stage.hms.harvard.edu/courses/course-v1:HarvardMedGlobalAcademy+ITA2+1T2018/course/

The last section 'Course Evaluation' is gated on 'CME Posttest'

Reply:
Could it be that the Webviews are rendering this differently.  Are the webviews not rendering the lock out the same as the web application view show to website users?

FYI: @nasthagiri

Reply:
@staubina Both the Mobile webviews and LTI consumers use the [render_xblock view](https://github.com/edx/edx-platform/blob/5750495f86cd83985b4793416e981cf91d52cdf6/lms/djangoapps/courseware/views/views.py#L1407).  And if you follow through that code, you'll notice that it eventually [calls has_access](https://github.com/edx/edx-platform/blob/57ba9b2ac6a1480ba61be483ebfbb19188429033/lms/djangoapps/courseware/module_render.py#L845), which eventually comes to this line of code.

Reply:
@BessieSteinberg thanks. I tested on Stage by going to the following link as a non-staff learner, and it looks like the access control check isn't working on Stage either:

https://globalacademy-stage.hms.harvard.edu/xblock/block-v1:HarvardMedGlobalAcademy+ITA2+1T2018+type@html+block@923db3ac860544dc934e605959f1661e

This is most surprising - and is worth debugging locally - by stepping through the code path I mentioned above to understand where the milestones checking is failing.  I guess I don't feel comfortable releasing Gated content changes without understanding why the content isn't actually Gated.  :)

Reply:
@nasthagiri: Isn't it generally the case that leaf node content doesn't apply access gating beyond simple enrollment checks (e.g. A/B tests)? I thought that some courses experimenting with adaptive learning (like Harvard's "Super-Earths") explicitly count on this behavior in order to work around certain platform limitations with dynamic content.

Reply:
It might just be that the milestone only registers the top level `block_id` and doesn't do the ancestor crawl?

Reply:
@nasthagiri 
So just tested this on my local devstack using master. It does seems like the access checks are being called as expected when going to a url like <site>/xblock/<usage-key> and I see an 404 Page Not Found error which is appropriate. Not sure why this isn't working on stage.. need to dig further.

Reply:
@bill-filler ok. thanks.  and you checked it on a leaf-node (like a problem-block or HTML-block) and not the sequential-node, right?

Reply:
@nasthagiri 
Sorry I was wrong. Was testing it on a Sequential. When I tried it on the Vertical (leaf-node) I was able to access.

Reply:
ok. then that confirms what @ormsbee is saying - current implementation doesn't enforce Gated content beyond the sequentials.

In that case, sorry for the detour here.  Y'all are right that this PR doesn't introduce any regressions in this regard.  So my comment here is no longer a blocker.